### ISSUE TITLE
hc-picklist sorts values alphabetically, even when it doesn't make sense to do so

### ISSUE THREAD

**Issue by isaaclyman** [HealthCatalyst/Fabric.Cashmere#1102]
**Summary**

When you provide values to an hc-picklist, it sorts them alphabetically. There doesn't seem to be a way to turn this off. Alphabetical sorting doesn't make sense for this picklist:

![image](https://user-images.githubusercontent.com/9139369/72112581-2b213c80-32fb-11ea-873a-2c1e1f0ce2c0.png)

It doesn't even make sense for this picklist, from the Cashmere examples page:

![image](https://user-images.githubusercontent.com/9139369/72112609-412efd00-32fb-11ea-901a-7a0eb6a41e0a.png)

No normal human lists the cardinal directions in ENSW order, that's ridiculous. I think people usually go NESW, or at least pick a point and proceed clockwise.

In short, I don't think we should be alphabetizing user input here _at all_. There's no warning that we're going to do this. At the very least, we should include an option in the modal config to turn off alphabetization.

**Comment by Br00t4L17y** [comment]
What if we had an option for `sort` which defaults to `none`, but could take `asc` or `desc` input?

**Comment by isaaclyman** [comment]
@Br00t4L17y that sounds good to me.

**Comment by andrew-frueh** [comment]
Yup - I like that idea too @Br00t4L17y.  @corykon I think you'll be happy to know that someone is actually using the PickList :-)  We've had designs on overhauling this component for a while.

Cory - if you have a minute to make this update, go for it.  If not I can take care of it.

**Comment by corykon** [comment]
Submitting a PR now. Sorry for the confusion. This component indeed needs a major overhaul, it was pretty quickly ported over from a very specific use case.

Would be great to find out if anyone is using this component with the "valueset" mode. If not, could cut out like 75% of the code.

**Comment by isaaclyman** [comment]
@corykon On the off chance my team is the only one using picklist, here's the sole reference to "valueset" in our codebase:

![image](https://user-images.githubusercontent.com/9139369/72373470-91280e00-36c5-11ea-8b95-44d008763abb.png)
